x-pack/osquerybeat/ext/osquery-extension/pkg/amcache/registry/registry.go

```diff
@@ -0,0 +1,249 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build windows
+
+package registry
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/forensicanalysis/fslib"
+	"github.com/forensicanalysis/fslib/systemfs"
+	"www.velocidex.com/golang/regparser"
+
+	"github.com/elastic/beats/v7/x-pack/osquerybeat/ext/osquery-extension/pkg/logger"
+)
+
+// ensureLogger ensures that a logger is provided. If no logger is provided,
+// a new logger is created and returned. A simple check to ensure
+// that the logger is not nil.
+func ensureLogger(log *logger.Logger) *logger.Logger {
+	if log == nil {
+		log = logger.New(os.Stderr, false)
+	}
+	return log
+}
+
+// getFileContents reads the contents of a file and returns it as a byte slice.
+// If the file is not readable, it will attempt to read it using a low level read
```

Comment:
this is just out of curiosity, I assume not using always fslib is for performance reasons?